modules/service/controllers/service/controller.go

```diff
@@ -47,17 +53,66 @@ type serviceHandler struct {
 
 	clusterDomainCache adminv1.ClusterDomainCache
 	publicDomainCache  adminv1.PublicDomainCache
+	configmaps         corev1controller.ConfigMapClient
 }
 
 func (s *serviceHandler) populate(service *riov1.Service, status riov1.ServiceStatus) ([]runtime.Object, riov1.ServiceStatus, error) {
 	if service.Spec.Template {
 		return nil, status, generic.ErrSkip
 	}
 
+	if err := s.ensureFeatures(service); err != nil {
+		return nil, status, err
+	}
+
 	os := objectset.NewObjectSet()
 	if err := populate.Service(service, os); err != nil {
 		return nil, status, err
 	}
 
 	return os.All(), status, nil
 }
+
+func (s *serviceHandler) ensureFeatures(service *riov1.Service) error {
+	cm, err := s.configmaps.Get(s.namespace, config.ConfigName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	conf, err := config.FromConfigMap(cm)
+	if err != nil {
+		return err
+	}
+
+	if services.AutoscaleEnable(service) && arch.IsAmd64() {
+		if conf.Features == nil {
+			conf.Features = map[string]config.FeatureConfig{}
+		}
+		f := conf.Features["autoscaling"]
+		f.Enabled = &[]bool{true}[0]
+		conf.Features["autoscaling"] = f
+	}
+
+	for _, con := range services.ToNamedContainers(service) {
+		if con.ImageBuild != nil && con.ImageBuild.Repo != "" && arch.IsAmd64() {
+			if conf.Features == nil {
+				conf.Features = map[string]config.FeatureConfig{}
+			}
+			f := conf.Features["build"]
+			f.Enabled = &[]bool{true}[0]
```

Comment:
I don't love this but... I think its the only way to do this in a single line....maybe https://stackoverflow.com/questions/28817992/how-to-set-bool-pointer-to-true-in-struct-literal